src/content/1.7/development/components/grid/tutorials/work-with-grid-actions.md

```diff
@@ -3,4 +3,180 @@ title: How to work with Grid Actions?
+                ->setOptions([
+                    'route' => 'admin_customers_export',
+                ])
+            )
+            ->add(
+                (new SimpleGridAction('common_refresh_list'))
+                ->setName($this->trans('Refresh list', [], 'Admin.Advparameters.Feature'))
+                ->setIcon('refresh')
+            )
+            ->add(
+                (new SimpleGridAction('common_show_query'))
+                ->setName($this->trans('Show SQL query', [], 'Admin.Actions'))
+                ->setIcon('code')
+            )
+            ->add(
+                (new SimpleGridAction('common_export_sql_manager'))
+                ->setName($this->trans('Export to SQL Manager', [], 'Admin.Actions'))
+                ->setIcon('storage')
+            );
+    }
+}
+```
+
+As you see, Grid stores Gird actions using a `GridActionCollection` object. This means that every action needed for Grid must be added to `GridActionCollection` using `add()` method.
+You'll find the list of existing actions in the documentation.
+
+## How to add Bulk actions to a Grid using a module?
+
+You can add additional actions to any Grid by hooking to the Grid workflow. You have to follow these steps:
+
+1. Register hook `action{gridId}GridDefinitionModifier` where `{gridId}` is Grid id (e.g. `customer` for Customers grid, this means that hook name would be `actionCustomerGridDefinitionModifier`).
+2. Implement hook method in your module:
+
+```php
+<?php
+// modules/my-module/my-module.php
+
+class Mymodule extends Module
+{
+    // ...
+
+    /**
+     * Use hook to add Bulk action for subscribing multiple customers to newsletter
+     */
+    public function hookActionCustomerGridDefinitionModifier(array $params)
+    {
+        // $params['definition'] is instance of \PrestaShop\PrestaShop\Core\Grid\Definition\GridDefinition
+        $params['definition']->getActions()->add(
+            ->add(
+                (new LinkGridAction('new_action'))
+                    ->setName($this->trans('New action', [], 'MyModule.Admin.Actions'))
+                    ->setIcon('add_circle') // icon from https://materializecss.com/icons.html by default
+                    ->setOptions([
+                        'route' => 'my_module_specific_route',
+                    ])
+            )
+        ;
+    }
+}
+```
+
+Now you should be able to see new action available into the Customers grid!
+
+## How to create custom Grid action?
+
+PrestaShop already comes with a bunch of actions that are available for use in your Grids or for extending PrestaShop grids.
+However, in some use cases you may find that the existing actions do not fit your needs.
+
+Luckily, there is a solution, you can create and register your own custom actions.
+
+First, you need to create a Grid action itself:
+
+```php
+// modules/mymodule/src/Grid/Action/Type/
+
+use PrestaShop\PrestaShop\Core\Grid\Action\AbstractGridAction;
+
+/** It extends AbstractGridAction,
```

Comment:
Replace this
```
/** It extends AbstractGridAction,
 * but you can also implement \PrestaShop\PrestaShop\Core\Grid\Action\GridActionInterface 
 */ if for some reason you want to avoid using the abstract class
```

by 
 ```
/**
 * It extends AbstractGridAction,
 * but you can also implement \PrestaShop\PrestaShop\Core\Grid\Action\GridActionInterface 
 * if for some reason you want to avoid using the abstract class
 */ 
```